### PR TITLE
Early gitsign support

### PR DESCRIPTION
**Pull request by adityasaky**

### CODE REVIEW

**Comment by adityasaky** [comment]
@wlynch I think this is ready for an early pass while I also work out how to test gitsign sigs without calling out to rekor.

File: internal/gitinterface/commit.go [hunk 1]
```diff
@@ -79,7 +86,60 @@ func VerifyCommitSignature(commit *object.Commit, key *tuf.Key) error {
 
 		return nil
 	case signerverifier.FulcioKeyType:
-		return ErrUnknownSigningMethod // TODO: implement
+		root, err := fulcioroots.Get()
+		if err != nil {
+			return errors.Join(ErrVerifyingSigstoreSignature, err)
```
Comment (wlynch) [inline internal/gitinterface/commit.go:91]: If you don't want to bump to Go 1.20 you could do something like this instead https://go.dev/play/p/L5P9FiIZS07

**Review by wlynch — APPROVED** [review]
Sigstore unit testing aside, this lgtm!

File: internal/gitinterface/commit.go (hunk 1 above)
Comment (adityasaky) [reply internal/gitinterface/commit.go:91]: Yeah, I think I'll keep 1.19 around for a bit longer.

**Pull request merged**